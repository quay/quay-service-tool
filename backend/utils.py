from flask_restful import Resource
from flask import request, make_response
from flask import current_app as app
from flask_login import current_user
from datetime import datetime
import logging
import json

logger = logging.getLogger(__name__)


def is_valid_severity(severity):
    return severity in ["default", "success", "info", "danger", "warning"]


class User(object):
    def __init__(self, is_authenticated=False, email=None, username=None, realm_access=None):
        self.email = email
        self.username = username
        self.is_authenticated = is_authenticated
        self.realm_access = realm_access


class Auth(Resource):
    def authenticate_user(token_info, config):
        if not token_info or not token_info.get("realm_access"):
            return User()

        realm_access = token_info["realm_access"]
        if not realm_access.get("roles") or not config["roles"] or \
           not (set(realm_access["roles"]) & set(config["roles"].values())):
            return User()

        return User(
            email=token_info["email"],
            username=token_info["name"],
            is_authenticated=True,
            realm_access=realm_access,
        )


def create_transaction(db):
    if type(db.obj).__name__ == "ObservableRetryingMySQLDatabase":
        try:
            db.close()
        except:
            pass

    return db.transaction()


def log_response(func):
    def wrapper(*args, **kwargs):
        response = func(*args, **kwargs)
        log_fn = logging.error
        if response.status_code == 500:
            log_fn = logging.exception
        if response.status_code == 200 or response.status_code == 201:
            log_fn = logging.info
        log_fn(
            f"{datetime.utcnow().strftime('%d %b, %Y, %H:%M:%S')} - "
            f"{request.method} - "
            f"{request.path} - "
            f"{response.status_code} - "
            f"{current_user.username} - "
            f"{current_user.email} - "
            f"{json.dumps(request.json)} - "
            f"{response.data}"
        )
        return response

    return wrapper


def verify_admin_permissions(func):
    def wrapper(*args, **kwargs):
        # Bypassing checks for local dev when auth is not on
        if app.config.get('is_local') and not app.config.get('test_auth'):
            return func(*args, **kwargs)

        if not current_user or not current_user.realm_access:
            return make_response(json.dumps({"message": "No RBAC defined for user"}), 401)

        ADMIN_ROLE = app.config.get('authentication', {}).get('roles', {}).get('ADMIN_ROLE')
        if ADMIN_ROLE not in current_user.realm_access.get('roles', None):
            return make_response(json.dumps({"message": "Require ADMIN permissions to perform action"}), 401)

        response = func(*args, **kwargs)
        return response
    return wrapper


def verify_export_compliance_permissions(func):
    def wrapper(*args, **kwargs):
        # Bypassing checks for local dev when auth is not on
        if app.config.get('is_local') and not app.config.get('test_auth'):
            return func(*args, **kwargs)

        if not current_user or not current_user.realm_access:
            return make_response(json.dumps({"message": "No RBAC defined for user"}), 401)

        EXPORT_COMPLIANCE_ROLE = app.config.get('authentication', {}).get('roles', {}).get('EXPORT_COMPLIANCE_ROLE')
        if EXPORT_COMPLIANCE_ROLE not in current_user.realm_access.get('roles', None):
            return make_response(json.dumps({"message": "Require EXPORT_COMPLIANCE_ROLE permissions to perform action"}), 401)

        response = func(*args, **kwargs)
        return response
    return wrapper


def verify_admin_or_export_perm(func):
    def wrapper(*args, **kwargs):
        # Bypassing checks for local dev when auth is not on
        if app.config.get('is_local') and not app.config.get('test_auth'):
            return func(*args, **kwargs)

        if not current_user or not current_user.realm_access:
            return make_response(json.dumps({"message": "No RBAC defined for user"}), 401)

        ADMIN_ROLE = app.config.get('authentication', {}).get('roles', {}).get('ADMIN_ROLE')
        EXPORT_COMPLIANCE_ROLE = app.config.get('authentication', {}).get('roles', {}).get('EXPORT_COMPLIANCE_ROLE')
        if not (set([ADMIN_ROLE, EXPORT_COMPLIANCE_ROLE]) & set(current_user.realm_access.get('roles', None))):
            return make_response(json.dumps({"message": f"Not enough permissions to perform action"}), 401)

        response = func(*args, **kwargs)
        return response
    return wrapper
