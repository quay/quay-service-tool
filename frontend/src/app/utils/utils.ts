export function accessibleRouteChangeHandler() {
  return window.setTimeout(() => {
    const mainContainer = document.getElementById('primary-app-container');
    if (mainContainer) {
      mainContainer.focus();
    }
  }, 50);
}

export function getErrorMessage(error){
  let message = "";
  if (error.response) {
    message = `Recieved HTTP status ${error.response.status} from server. Response: ${error.response.data.message}`;
  } else if (error.request) {
    message = "Unable to get response from server";
  } else {
    message = "Error initializing request";
  }
  return message
}
