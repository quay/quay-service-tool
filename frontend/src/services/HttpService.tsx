import axios from "axios";
import UserService from "./UserService";

const HttpMethods = {
  GET: 'GET',
  POST: 'POST',
  DELETE: 'DELETE',
};

const _axios = axios.create();

const configure = () => {
  _axios.interceptors.request.use((config) => {
    if (process.env.NODE_ENV !== "production") {
      return config;
    }
    else if (UserService.isLoggedIn()) {
      const cb = () => {
        config.headers.Authorization = `Bearer ${UserService.getToken()}`;
        return Promise.resolve(config);
      };
      return UserService.updateToken(cb);
    }
  },function(error) {
    return Promise.reject(error);
  });
};

const axiosClient = _axios;

const HttpService = {
  HttpMethods,
  configure,
  axiosClient,
};

export default HttpService;
