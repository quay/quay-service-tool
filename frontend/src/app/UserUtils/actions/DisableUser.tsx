import * as React from 'react';
import {
    ActionGroup,
    Alert,
    AlertActionCloseButton,
    Button,
    Card,
    CardBody,
    CardTitle,
    Form,
    FormGroup,
    Modal,
    ModalVariant,
    PageSection,
    TextInput,
  } from '@patternfly/react-core';
import HttpService from "../../../services/HttpService";
import { useState } from 'react';
import { getErrorMessage } from '@app/utils/utils';

type Props = {

};

type Alert = {
    variant: "success" | "danger" | "default" | "warning" | "info" | undefined;
    show: boolean;
    title: string;
}

export const DisableUser: React.FunctionComponent = (props: Props) => {
    const [alert, setAlert] = useState<Alert>({
        variant: 'success',
        show: false,
        title: ''
    });
    const [username, setUsername] = useState('');
    const [showModal, setShowModal] = useState(false);

    async function promptDisableUser(){
        if(username.length == 0){
            setAlert({ variant: 'danger', show: true, title: 'Please enter a username' });
            return;
        }
        HttpService.axiosClient.get(`/user/${username}`)
            .then(function(response){
                if(!response.data.enabled){
                    setAlert({variant: "danger", show: true, title: `User ${username} is already disabled`});
                } else {
                    setShowModal(true);
                }
            })
            .catch((error) => { 
                let errMessage = error.response.status == 404 ? `User ${username} does not exist` : getErrorMessage(error);
                setAlert({variant: "danger", show: true, title: errMessage}) 
            });
    }

    async function disableUser(){
        HttpService.axiosClient.put(`/user/${username}?enable=false`)
          .then(function(response){
            setShowModal(false);
            setAlert({variant: "success", show: true, title: `User ${response.data.user} disabled`});
          })
          .catch((error) => { 
            setShowModal(false);  
            setAlert({variant: "danger", show: true, title: getErrorMessage(error)}) 
        });
    }

    let modal: JSX.Element | null = null;
    if(showModal){
        let title = `Disable user ${username}?`;
        modal = (
            <Modal
            title={title}
            isOpen={showModal}
            variant={ModalVariant.small}
            aria-label="feedback modal"
            showClose={true}
            onClose={() => setShowModal(false)}
            aria-describedby="no-header-example"
            >
                <Button id="disable-user-confirm" variant="primary" onClick={disableUser}>Disable User</Button>
            </Modal>
        );
    }
    return (
    <PageSection>
        {modal}
        <Card>
          <CardTitle className={"text-uppercase"}> Disable User </CardTitle>
          {alert.show && <Alert id="disable-user-alert" isInline actionClose={<AlertActionCloseButton onClose={() => setAlert({ variant: 'success', show: false, title: ''})} />} variant={alert.variant} title={alert.title} />}
          <CardBody>
            <Form id="enable-user-form">
              <FormGroup label="Username" fieldId="disable-username" isRequired>
                <TextInput
                  isRequired
                  type="text"
                  id="disable-username"
                  name="disable-username"
                  value={username}
                  onChange={(value) => setUsername(value)}
                  placeholder="Username"
                />
              </FormGroup>

              <ActionGroup>
                <Button id="disable-user-submit" variant="primary" onClick={promptDisableUser}>Disable User</Button>
              </ActionGroup>
            </Form>
          </CardBody>
        </Card>
      </PageSection>
    );
}
