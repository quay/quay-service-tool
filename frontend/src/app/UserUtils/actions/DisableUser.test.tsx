import { mount, shallow } from 'enzyme';
import React from 'react';
import axios from "axios";
import { DisableUser } from './DisableUser';
import { List, Modal } from '@patternfly/react-core';
import HttpService from "../../../services/HttpService";
import { mocked } from 'ts-jest/utils';
import {act} from 'react-dom/test-utils';

jest.mock('../../../services/HttpService', () => ({
    axiosClient: {
        get: jest.fn(),
        put: jest.fn()
    }
}));

// Wait for all pending promises to resolve
// Adds promise to end of event loop and waits for it to resolve
function flushPromises() {
    return new Promise(resolve => setImmediate(resolve));
}

async function typeUsername(view, username){
    let input = view.find("input#disable-username").at(0);
    await act(async () => {
        input.getDOMNode().setAttribute('value', username);
        input.simulate('change', {currentTarget: input});
    });
}

async function typeQueue(view, queue){
    let queueInput = view.find("input#queue").at(0);
    await act(async () => {
        queueInput.getDOMNode().setAttribute('value', queue);
        queueInput.simulate('change', {currentTarget: queueInput});
    });
}

// Clicking button triggers other async requests and updates state
async function asyncClickButton(view, button){
    await act(async () => {
        button.simulate('click');
        await flushPromises(); // Pauses execution of tests until promises in the component have been resolved
        view.update(); // Sync enzyme component tree with the react component tree
    });
}

describe('Disable users tests', ()=>{
    it('should render', () => {
        const view = mount(<DisableUser />);
        expect(view).toMatchSnapshot();
    });

    it('Should alert error if username is not provided', () => {
        const view = mount(<DisableUser />);
        view.find('button#disable-user-submit').simulate('click');
        expect(view.find('Alert#disable-user-alert').props()).toHaveProperty('variant', 'danger');
        expect(view.find('Alert#disable-user-alert').text().includes('Please enter a username')).toBe(true);
    })

    it('Should alert error if queue is not provided', async () => {
        const view = mount(<DisableUser />);
        await typeUsername(view, 'test');
        view.find('button#disable-user-submit').simulate('click');
        expect(view.find('Alert#disable-user-alert').props()).toHaveProperty('variant', 'danger');
        expect(view.find('Alert#disable-user-alert').text().includes('Please enter a queue name')).toBe(true);
    })

    it('Should alert if given username isn\'t found in backend', async () => {
        let username = 'nonexistentusername';
        mocked(HttpService, true).axiosClient.get.mockRejectedValue({response: {status: 404, data:{message: `Could not find user ${username}`}}});
        const view = mount(<DisableUser />);
        await typeUsername(view, username);
        await typeQueue(view, 'testqueue');
        await asyncClickButton(view, view.find('button#disable-user-submit'));
        expect(view.find('Alert#disable-user-alert').props()).toHaveProperty('variant', 'danger');
        expect(view.find('Alert#disable-user-alert').text().includes(`User ${username} does not exist`)).toBe(true);
    })

    it('Should alert if user is already disabled', async ()=>{
        mocked(HttpService, true).axiosClient.get.mockResolvedValue({data: {enabled: false}});
        let username = "existingusername";
        const view = mount(<DisableUser />);
        await typeUsername(view, username);
        await typeQueue(view, 'testqueue');
        await asyncClickButton(view, view.find('button#disable-user-submit'));
        expect(view.find('Alert#disable-user-alert').props()).toHaveProperty('variant', 'danger');
        expect(view.find('Alert#disable-user-alert').text().includes(`User ${username} is already disabled`)).toBe(true);
    })

    it('Should disable user', async()=>{
        let username = "existinguser";
        mocked(HttpService, true).axiosClient.get.mockResolvedValue({data: {user: username, enabled: true}});
        mocked(HttpService, true).axiosClient.put.mockResolvedValue({data: {message: "User updated successfully", user: "existinguser",enabled: false}});
        const view = mount(<DisableUser />);
        await typeUsername(view, username);
        await typeQueue(view, 'testqueue');
        await asyncClickButton(view, view.find('button#disable-user-submit'));
        expect(view.find('Modal').props()).toHaveProperty('isOpen', true);
        expect(view.find('Modal').props()).toHaveProperty('title', `Disable user ${username}?`);
        await asyncClickButton(view, view.find('button#disable-user-confirm'));
        expect(view.find('Alert#disable-user-alert').props()).toHaveProperty('variant', 'success');
        expect(view.find('Alert#disable-user-alert').text().includes(`User ${username} disabled`)).toBe(true);
        expect(view.find('Modal').exists()).toBe(false);
    })
});
