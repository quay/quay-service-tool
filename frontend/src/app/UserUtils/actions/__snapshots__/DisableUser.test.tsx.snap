// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disable users tests should render 1`] = `
<DisableUser>
  <PageSection>
    <section
      className="pf-c-page__main-section"
    >
      <Card>
        <article
          className="pf-c-card"
          data-ouia-component-id="OUIA-Generated-Card-1"
          data-ouia-component-type="PF4/Card"
          data-ouia-safe={true}
          id=""
        >
          <CardTitle
            className="text-uppercase"
          >
            <div
              className="pf-c-card__title text-uppercase"
            >
               Disable User and Clean Build Queue 
            </div>
          </CardTitle>
          <CardBody>
            <div
              className="pf-c-card__body"
            >
              <Form
                id="enable-user-form"
              >
                <form
                  className="pf-c-form"
                  id="enable-user-form"
                  noValidate={true}
                >
                  <FormGroup
                    fieldId="disable-username"
                    isRequired={true}
                    label="Username"
                  >
                    <div
                      className="pf-c-form__group"
                    >
                      <div
                        className="pf-c-form__group-label"
                      >
                        <label
                          className="pf-c-form__label"
                          htmlFor="disable-username"
                        >
                          <span
                            className="pf-c-form__label-text"
                          >
                            Username
                          </span>
                          <span
                            aria-hidden="true"
                            className="pf-c-form__label-required"
                          >
                             
                            *
                          </span>
                        </label>
                         
                      </div>
                      <div
                        className="pf-c-form__group-control"
                      >
                        <TextInput
                          id="disable-username"
                          isRequired={true}
                          name="disable-username"
                          onChange={[Function]}
                          placeholder="Username"
                          type="text"
                          value=""
                        >
                          <TextInputBase
                            aria-label={null}
                            className=""
                            id="disable-username"
                            innerRef={null}
                            isDisabled={false}
                            isLeftTruncated={false}
                            isReadOnly={false}
                            isRequired={true}
                            name="disable-username"
                            onChange={[Function]}
                            ouiaSafe={true}
                            placeholder="Username"
                            type="text"
                            validated="default"
                            value=""
                          >
                            <input
                              aria-invalid={false}
                              aria-label={null}
                              className="pf-c-form-control"
                              data-ouia-component-id="OUIA-Generated-TextInputBase-1"
                              data-ouia-component-type="PF4/TextInput"
                              data-ouia-safe={true}
                              disabled={false}
                              id="disable-username"
                              name="disable-username"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              placeholder="Username"
                              readOnly={false}
                              required={true}
                              type="text"
                              value=""
                            />
                          </TextInputBase>
                        </TextInput>
                      </div>
                    </div>
                  </FormGroup>
                  <FormGroup
                    fieldId="queue"
                    isRequired={true}
                    label="queue"
                  >
                    <div
                      className="pf-c-form__group"
                    >
                      <div
                        className="pf-c-form__group-label"
                      >
                        <label
                          className="pf-c-form__label"
                          htmlFor="queue"
                        >
                          <span
                            className="pf-c-form__label-text"
                          >
                            queue
                          </span>
                          <span
                            aria-hidden="true"
                            className="pf-c-form__label-required"
                          >
                             
                            *
                          </span>
                        </label>
                         
                      </div>
                      <div
                        className="pf-c-form__group-control"
                      >
                        <TextInput
                          id="queue"
                          isRequired={true}
                          name="queue"
                          onChange={[Function]}
                          placeholder="Queue"
                          type="text"
                          value=""
                        >
                          <TextInputBase
                            aria-label={null}
                            className=""
                            id="queue"
                            innerRef={null}
                            isDisabled={false}
                            isLeftTruncated={false}
                            isReadOnly={false}
                            isRequired={true}
                            name="queue"
                            onChange={[Function]}
                            ouiaSafe={true}
                            placeholder="Queue"
                            type="text"
                            validated="default"
                            value=""
                          >
                            <input
                              aria-invalid={false}
                              aria-label={null}
                              className="pf-c-form-control"
                              data-ouia-component-id="OUIA-Generated-TextInputBase-2"
                              data-ouia-component-type="PF4/TextInput"
                              data-ouia-safe={true}
                              disabled={false}
                              id="queue"
                              name="queue"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              placeholder="Queue"
                              readOnly={false}
                              required={true}
                              type="text"
                              value=""
                            />
                          </TextInputBase>
                        </TextInput>
                      </div>
                    </div>
                  </FormGroup>
                  <ActionGroup>
                    <div
                      className="pf-c-form__group pf-m-action"
                    >
                      <div
                        className="pf-c-form__group-control"
                      >
                        <div
                          className="pf-c-form__actions"
                        >
                          <Button
                            id="disable-user-submit"
                            onClick={[Function]}
                            variant="primary"
                          >
                            <ButtonBase
                              id="disable-user-submit"
                              innerRef={null}
                              onClick={[Function]}
                              variant="primary"
                            >
                              <button
                                aria-disabled={false}
                                aria-label={null}
                                className="pf-c-button pf-m-primary"
                                data-ouia-component-id="OUIA-Generated-Button-primary-1"
                                data-ouia-component-type="PF4/Button"
                                data-ouia-safe={true}
                                disabled={false}
                                id="disable-user-submit"
                                onClick={[Function]}
                                role={null}
                                type="button"
                              >
                                Disable User
                              </button>
                            </ButtonBase>
                          </Button>
                        </div>
                      </div>
                    </div>
                  </ActionGroup>
                </form>
              </Form>
            </div>
          </CardBody>
        </article>
      </Card>
    </section>
  </PageSection>
</DisableUser>
`;

exports[`Enable users tests should render 1`] = `
<DisableUser>
  <PageSection>
    <section
      className="pf-c-page__main-section"
    >
      <Card>
        <article
          className="pf-c-card"
          data-ouia-component-id="OUIA-Generated-Card-1"
          data-ouia-component-type="PF4/Card"
          data-ouia-safe={true}
          id=""
        >
          <CardTitle
            className="text-uppercase"
          >
            <div
              className="pf-c-card__title text-uppercase"
            >
               Disable User and Clean Build Queue 
            </div>
          </CardTitle>
          <CardBody>
            <div
              className="pf-c-card__body"
            >
              <Form
                id="enable-user-form"
              >
                <form
                  className="pf-c-form"
                  id="enable-user-form"
                  noValidate={true}
                >
                  <FormGroup
                    fieldId="disable-username"
                    isRequired={true}
                    label="Username"
                  >
                    <div
                      className="pf-c-form__group"
                    >
                      <div
                        className="pf-c-form__group-label"
                      >
                        <label
                          className="pf-c-form__label"
                          htmlFor="disable-username"
                        >
                          <span
                            className="pf-c-form__label-text"
                          >
                            Username
                          </span>
                          <span
                            aria-hidden="true"
                            className="pf-c-form__label-required"
                          >
                             
                            *
                          </span>
                        </label>
                         
                      </div>
                      <div
                        className="pf-c-form__group-control"
                      >
                        <TextInput
                          id="disable-username"
                          isRequired={true}
                          name="disable-username"
                          onChange={[Function]}
                          placeholder="Username"
                          type="text"
                          value=""
                        >
                          <TextInputBase
                            aria-label={null}
                            className=""
                            id="disable-username"
                            innerRef={null}
                            isDisabled={false}
                            isLeftTruncated={false}
                            isReadOnly={false}
                            isRequired={true}
                            name="disable-username"
                            onChange={[Function]}
                            ouiaSafe={true}
                            placeholder="Username"
                            type="text"
                            validated="default"
                            value=""
                          >
                            <input
                              aria-invalid={false}
                              aria-label={null}
                              className="pf-c-form-control"
                              data-ouia-component-id="OUIA-Generated-TextInputBase-1"
                              data-ouia-component-type="PF4/TextInput"
                              data-ouia-safe={true}
                              disabled={false}
                              id="disable-username"
                              name="disable-username"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              placeholder="Username"
                              readOnly={false}
                              required={true}
                              type="text"
                              value=""
                            />
                          </TextInputBase>
                        </TextInput>
                      </div>
                    </div>
                  </FormGroup>
                  <FormGroup
                    fieldId="queue"
                    isRequired={true}
                    label="queue"
                  >
                    <div
                      className="pf-c-form__group"
                    >
                      <div
                        className="pf-c-form__group-label"
                      >
                        <label
                          className="pf-c-form__label"
                          htmlFor="queue"
                        >
                          <span
                            className="pf-c-form__label-text"
                          >
                            queue
                          </span>
                          <span
                            aria-hidden="true"
                            className="pf-c-form__label-required"
                          >
                             
                            *
                          </span>
                        </label>
                         
                      </div>
                      <div
                        className="pf-c-form__group-control"
                      >
                        <TextInput
                          id="queue"
                          isRequired={true}
                          name="queue"
                          onChange={[Function]}
                          placeholder="Queue"
                          type="text"
                          value=""
                        >
                          <TextInputBase
                            aria-label={null}
                            className=""
                            id="queue"
                            innerRef={null}
                            isDisabled={false}
                            isLeftTruncated={false}
                            isReadOnly={false}
                            isRequired={true}
                            name="queue"
                            onChange={[Function]}
                            ouiaSafe={true}
                            placeholder="Queue"
                            type="text"
                            validated="default"
                            value=""
                          >
                            <input
                              aria-invalid={false}
                              aria-label={null}
                              className="pf-c-form-control"
                              data-ouia-component-id="OUIA-Generated-TextInputBase-2"
                              data-ouia-component-type="PF4/TextInput"
                              data-ouia-safe={true}
                              disabled={false}
                              id="queue"
                              name="queue"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              placeholder="Queue"
                              readOnly={false}
                              required={true}
                              type="text"
                              value=""
                            />
                          </TextInputBase>
                        </TextInput>
                      </div>
                    </div>
                  </FormGroup>
                  <ActionGroup>
                    <div
                      className="pf-c-form__group pf-m-action"
                    >
                      <div
                        className="pf-c-form__group-control"
                      >
                        <div
                          className="pf-c-form__actions"
                        >
                          <Button
                            id="disable-user-submit"
                            onClick={[Function]}
                            variant="primary"
                          >
                            <ButtonBase
                              id="disable-user-submit"
                              innerRef={null}
                              onClick={[Function]}
                              variant="primary"
                            >
                              <button
                                aria-disabled={false}
                                aria-label={null}
                                className="pf-c-button pf-m-primary"
                                data-ouia-component-id="OUIA-Generated-Button-primary-1"
                                data-ouia-component-type="PF4/Button"
                                data-ouia-safe={true}
                                disabled={false}
                                id="disable-user-submit"
                                onClick={[Function]}
                                role={null}
                                type="button"
                              >
                                Disable User
                              </button>
                            </ButtonBase>
                          </Button>
                        </div>
                      </div>
                    </div>
                  </ActionGroup>
                </form>
              </Form>
            </div>
          </CardBody>
        </article>
      </Card>
    </section>
  </PageSection>
</DisableUser>
`;
